### DISCUSSION TITLE
Welcome to concrete-properties Discussions!

### DISCUSSION THREAD

**Discussion by robbievanleeuwen** [robbievanleeuwen/concrete-properties#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ahmetanildindar** [comment]
Dear @robbievanleeuwen,
This is a very good repository for whom intending to foster the ideas and efforts in analyzing the RC section properties. I'll share the repo to my students so that we can support the codes in terms of development and examples. 

All the best from Istanbul. 

++Ahmet Anil Dindar

**Reply by robbievanleeuwen** [reply]
Hi @ahmetanildindar thanks for the kind words! Keen to hear any feedback you or your students have with *concretepropeties*!

**Comment by afugur** [comment]
hi robbie,

First of all, very nice work. I have a question for you. I want to assign nonlinear material to the inner region of a section with a concrete cover and linear material to the outer region. Is there a way to do this?

I tried this 

```
conc_linear = Concrete(
    name="Linear Concrete",
    density=2.4e-6,
    stress_strain_profile=ssp.ConcreteLinear(
        elastic_modulus=35e3, ultimate_strain=0.0035
    ),
    ultimate_stress_strain_profile=ssp.BilinearStressStrain(
        compressive_strength=40,
        compressive_strain=0.00175,
        ultimate_strain=0.0035,
    ),
    alpha_squash=0.85,
    flexural_tensile_strength=3.5,
    colour="red",
)

conc_nonlinear = Concrete(
    name="Non-Linear Concrete",
    density=2.4e-6,
    stress_strain_profile=ssp.EurocodeNonLinear(
        elastic_modulus=35e3,
        ultimate_strain=0.0035,
        compressive_strength=40,
        compressive_strain=0.0023,
        tensile_strength=3.5,
        tension_softening_stiffness=10e3,
    ),
    ultimate_stress_strain_profile=ssp.BilinearStressStrain(
        compressive_strength=40,
        compressive_strain=0.00175,
        ultimate_strain=0.0035,
    ),
    alpha_squash=0.85,
    flexural_tensile_strength=3.5,
    colour="blue",
)
```
```
steel_ep = Steel(
    name="Steel - Elastic-Plastic",
    density=7.85e-6,
    stress_strain_profile=ssp.SteelElasticPlastic(
        yield_strength=500,
        elastic_modulus=200e3,
        fracture_strain=0.05,
    ),
    colour="grey",
)
```
```
col = rectangular_section(d=350, b=600,material=conc_nonlinear)
col1 = rectangular_section(d=450, b=700,material=conc_linear).align_center(align_to=col)

col2 = col1 + col
# add bars to column
geom = add_bar_rectangular_array(
    geometry=col2,
    area=450,
    material=steel_ep,
    n_x=6,
    x_s=492 / 5,
    n_y=3,
    y_s=121,
    anchor=(54, 54),
    exterior_only=True,
)
```

then I got this.

![image](https://user-images.githubusercontent.com/62245496/182448800-c62959bb-6557-43c8-a297-7750a7ff54be.png)

Is it true ?

**Reply by connorferster** [reply]
Hi @afugur,

From the plot, the section _looks_ good however it seems that you have overlapping regions which are not permitted in concreteproperties. See below.

```python
col = rectangular_section(d=350, b=600,material=conc_nonlinear)
col1 = rectangular_section(d=450, b=700,material=conc_linear).align_center(align_to=col)

col2 = col1 + col # Ambiguity created here because these sections overlap
```

The plot we see is showing the materials as they are understood based on the assigned control points in each region however if I run the following after running the script you provided:

![image](https://user-images.githubusercontent.com/38441489/182458937-1e2a0e29-62ae-4b4f-bafa-2cd9f953668c.png)

I can see that the center concrete geometry is assigned two control points (`0` and `1`). This shows that there are overlapping regions. 

To prevent overlapping regions, we need to use the following modelling technique:

```python
col2 = (col1 - col) + col
```

Then the control points of the concrete regions are assigned correctly and there are no more overlapping regions:

![image](https://user-images.githubusercontent.com/38441489/182459290-79d1ac19-0eb3-40f0-9590-d602dc273539.png)

You can read more about this, here:
https://sectionproperties.readthedocs.io/en/latest/rst/advanced_geom.html

**Reply by afugur** [reply]
Thank you so much.

**Reply by robbievanleeuwen** [reply]
Thanks @connorferster great explanation!

@afugur you should also get a warning if you try to create a ``ConcreteSection`` object with geometry that has overlapping regions. Running your code I get this warning:

```shell
UserWarning: The provided geometry contains overlapping regions, results may be incorrect.
```

Using ``col2 = (col1 - col) + col`` removes this warning.

**Reply by afugur** [reply]
Thanks @robbievanleeuwen,

I saw the "UserWarning: The provided geometry contains overlapping regions, results may be incorrect." problem. Because of that I wanted to know true way :)
